generate_smr_inputs/1.0/generate_smr_inputs.R

```diff
@@ -16,107 +17,111 @@
-parser$add_argument("--capture", "-c", nargs=1, type= 'character', help="Path to the capture--projection/all--{projection}.seg file")
-parser$add_argument("--projection", "-p", nargs=1, type= 'character', required=TRUE, help="Genome build projection. e.g. hg38 or grch37")
-parser$add_argument("--output_dir", "-o", nargs=1, type= 'character', required=TRUE, help="Path to write the combined seg file for the case set")
-parser$add_argument("--all_sample_sets", nargs=1, type= 'character', required=TRUE, help="Tab delimited file where the first column is sample ID
-                                                        and the rest of the columns are named after case sets.
-                                                        Samples will have a 1 in a column if they are part of that case set,
-                                                        and zero otherwise.")
-parser$add_argument("--case_set", nargs=1, required=TRUE, help="Name of the case set to subset the region data to. Must match a column name in ALL_SAMPLE_SETS")
-
-# Gets the args as a named list
-args <- parser$parse_args()
-
-# Check that at least one of -g or -c is given
-if ( (length(args$genome)==0) && (length(args$capture)==0) ){
-  stop(paste("Exiting because at least one genome or capture seg file was not provided. \n"))
-}
+# Input snakemake variables
+subsetting_categories_file <- snakemake@input[["subsetting_categories"]]
 
-# Check existance of sample set file -----------------------------------------------------
-if (file.exists(args$all_sample_sets)) {
-  full_case_set = suppressMessages(read_tsv(args$all_sample_sets))
-} else {
-  stop(paste("Exiting because sample sets file", args$all_sample_sets, "is not found. \n"))
-}
+case_set <- snakemake@wildcards[["case_set"]]
+projection <- snakemake@wildcards[["projection"]]
+launch_date <- snakemake@wildcards[["launch_date"]]
 
-full_case_set =
-  full_case_set %>% rename_at(vars(matches(
-    "sample_id", ignore.case = TRUE
-  )),
-  ~ "ID")
-
-# Get sample IDs of the case_set
-case_set_samples <-full_case_set %>%
-  dplyr::filter(!!sym(args$case_set) == 1) %>%
-  pull(ID)
-
-# Load genome seg file and get regions for the caseset -------------------
-if (length(args$genome) != 0){ # if -g value providced
-  cat("Loading genome seg... \n")
-  if (!file.exists(args$genome)) {
-    stop(paste("Exiting because genome data seg file", args$genome, "is not found. \n"))
-  } else {
-    cat("Finding available data for samples in requested case set... \n")
-    genome_seg <- suppressMessages(read_tsv(args$genome, col_types = cols())) %>%
-      filter(ID %in% case_set_samples)
-  }
-}
+output_dir <- snakemake@config[["lcr-modules"]][["gistic2"]][["dirs"]][["prepare_seg"]]
+
+seq_type <- unlist(snakemake@params[["seq_type"]])
+metadata_str <- snakemake@params[["metadata"]]
+
+metadata <- data.frame(sample_id=metadata_str[c(TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE)],
```

Comment:
Just wonder what this does? Seems like it makes the data frame with a diagonal of TRUE for each column but not sure what it does later on

Reply:
Yes, it's not intuitive. When the pandas dataframe is passed to this R script from snakemake, it's defined as a list of lists I believe. This section converts it to a data.frame objects, where the "TRUE" and "FALSE" are used to define the columns from the list of lists object.

Reply:
I will add a comment to the code explaining this

Reply:
AH got it, thanks!!

Reply:
This metadata table is the PIPELINE_SAMPLES from snakemake, so it works in place of reading in "data/metadata/gambl_samples_available.tsv". This way, the pipeline does not get triggered to rerun if that samples available tsv is updated, since it isn't directly relying on it as an input file. And also, in the case where users need to specify specific samples in way that the level3_subsetting_categories.tsv doesn't cover, it can be done through this dataframe.